lib/Server.js

```diff
@@ -937,19 +937,6 @@ class Server {
       }
     }
 
-    // also allow public hostname if provided
-    if (typeof this.options.public === 'string') {
-      const idxPublic = this.options.public.indexOf(':');
-      const publicHostname =
-        idxPublic >= 0
-          ? this.options.public.substr(0, idxPublic)
-          : this.options.public;
-
-      if (hostname === publicHostname) {
-        return true;
-      }
-    }
```

Comment:
We have mixed logic for proxied server and socket url, it is weird, when you proxy dev server you need specify `host` in `firewall`